core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -101,12 +108,30 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  /**
+   * Set the checksum that the checksumOutputStream should use
+   */
+  def setChecksum(checksum: Checksum): Unit = {
+    if (checksumOutputStream == null) {
+      this.checksumEnabled = true
+      this.checksum = checksum
+    } else {
+      checksumOutputStream.setChecksum(checksum)
```

Comment:
So `checksumOutputStream` is not null after DiskBlockObjectWriter is initialized. Is the intention here to change the checksum even after this initialization?

Reply:
Yes, it's intentional. In the case of `ShuffleExternalSorter` spill, one `DiskBlockObjectWriter` would serve multiple partitions and different partitions should use different checksums.